### ISSUE TITLE
Docker-compose: Parsed but still unable to fread the specific log file 'goaccess'

### ISSUE THREAD

**Issue by renegadevi** [allinurl/goaccess#2247]
GoAccess seem to find and parse the log with >34k entries, but then gives a fatal error because it's unable to "**fread**" a log file 'goaccess'?

I found a previous issue (https://github.com/allinurl/goaccess/issues/2023) which had the issue of not even parsing the log, but even with the correct path for, it's still error's out after it's been parsed?

docker-compose.yml snippet
```yaml
...
nginx:
    ...
    volumes:
      - ./services/web:/home/app/web
      - ./services/web/app/static:/home/app/web/app/static
      - ./services/nginx/logs:/var/log/nginx
    ports:
      - 80:80
      - 443:443
      - 8080:8080
    restart: always

goaccess:
    image: allinurl/goaccess:latest
    ports:
      - 7890:7890
    volumes:
      - ./services/nginx/logs:/opt/log
      - ./services/goaccess:/goaccess
      - ./services/goaccess/html/index.html:/html/index.html
    command: ["goaccess", "--no-global-config", "--config-file=/goaccess/goaccess.conf"]
    depends_on:
      - nginx
...
```

goaccess.conf
```
log-file /opt/log/access.log
output /html/index.html
```

output
```
[PARSING /opt/log/access.log] {34729} @ {34729/s}
GoAccess - version 1.5.3 - Dec 12 2021 01:13:22
Config file: /goaccess/goaccess.conf

Fatal error has occurred
Error occurred at: src/parser.c - set_initial_persisted_data - 1964
Unable to fread the specified log file 'goaccess'

exit code: 1
```

**Comment by allinurl** [comment]
Looks like a path issue. Have you tried running it as (please change the paths accordingly):

    $ echo 'log-format VCOMBINED' > /home/user/goaccess.conf
    $ sudo mkdir /srv
    $ sudo cat access.log | docker run -v /home/user/goaccess.conf:/srv/goaccess.conf --rm -i -e LANG=$LANG allinurl/goaccess -a -o html -p /srv/goaccess.conf - > report.html

**Comment by renegadevi** [comment]
> Looks like a path issue. Have you tried running it as (please change the paths accordingly):
> 
> ```
> $ echo 'log-format VCOMBINED' > /home/user/goaccess.conf
> $ sudo mkdir /srv
> $ sudo cat access.log | docker run -v /home/user/goaccess.conf:/srv/goaccess.conf --rm -i -e LANG=$LANG allinurl/goaccess -a -o html -p /srv/goaccess.conf - > report.html
> ```

Yes, that works outside... but once it's in docker-compose, VCOMBINED does not work.

```
 [PARSING goaccess] {0} @ {0/s}
==1== GoAccess - Copyright (C) 2009-2020 by Gerardo Orellana
==1== https://goaccess.io - <hello@goaccess.io>
==1== Released under the MIT License.
==1==
==1== FILE: /srv/access.log
==1== Parsed 10 lines producing the following errors:
==1==
==1== Token '+0000]' doesn't match specifier '%h'
==1== Token '+0000]' doesn't match specifier '%h'
==1== Token '+0000]' doesn't match specifier '%h'
==1== Token '+0000]' doesn't match specifier '%h'
==1== Token '+0000]' doesn't match specifier '%h'
==1== Token '+0000]' doesn't match specifier '%h'
==1== Token '+0000]' doesn't match specifier '%h'
==1== Token '+0000]' doesn't match specifier '%h'
==1== Token '+0000]' doesn't match specifier '%h'
==1== Token '+0000]' doesn't match specifier '%h'
==1==
==1== Format Errors - Verify your log/date/time format
exit code: 1
```
but still when I change it to `log-format %h %^[%d:%t %^] "%m %U" %s %b "%R" "%u" %^` I get same issue as before. I can make it work by using your command by generating a .html page that I can access from the web, but then it's not running behind the container/port like it should.

**Comment by allinurl** [comment]
Could you please post a few lines from your access log?

**Comment by renegadevi** [comment]
```
10.0.2.100 - - [12/Dec/2021:15:49:12 +0000] "GET / HTTP/1.1" 200 58 "-" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10.15; rv:94.0) Gecko/20100101 Firefox/94.0"
10.0.2.100 - - [12/Dec/2021:15:49:12 +0000] "GET / HTTP/1.1" 200 58 "-" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10.15; rv:94.0) Gecko/20100101 Firefox/94.0"
10.0.2.100 - - [12/Dec/2021:15:49:13 +0000] "GET / HTTP/1.1" 200 58 "-" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10.15; rv:94.0) Gecko/20100101 Firefox/94.0"
10.0.2.100 - - [12/Dec/2021:15:49:13 +0000] "GET / HTTP/1.1" 200 58 "-" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10.15; rv:94.0) Gecko/20100101 Firefox/94.0"
10.0.2.100 - - [12/Dec/2021:15:49:13 +0000] "GET / HTTP/1.1" 200 58 "-" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10.15; rv:94.0) Gecko/20100101 Firefox/94.0"
10.0.2.100 - - [12/Dec/2021:15:49:13 +0000] "GET / HTTP/1.1" 200 58 "-" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10.15; rv:94.0) Gecko/20100101 Firefox/94.0"
10.0.2.100 - - [12/Dec/2021:15:49:13 +0000] "GET / HTTP/1.1" 200 58 "-" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10.15; rv:94.0) Gecko/20100101 Firefox/94.0"
10.0.2.100 - - [12/Dec/2021:22:03:51 +0000] "GET / HTTP/1.1" 200 58 "-" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10.15; rv:94.0) Gecko/20100101 Firefox/94.0"
10.0.2.100 - - [12/Dec/2021:22:03:51 +0000] "GET /favicon.ico HTTP/1.1" 200 32988 "https://192.168.199.6/" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10.15; rv:94.0) Gecko/20100101 Firefox/94.0"
10.0.2.100 - - [16/Dec/2021:22:09:23 +0000] "GET / HTTP/1.1" 200 58 "-" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10.15; rv:94.0) Gecko/20100101 Firefox/94.0"
```

**Comment by renegadevi** [comment]
Alright, I've got it working now. The main issue was the CMD/command. If I ignored the command, the man page showed up and it gave me the hint of maybe it was trying to parse it as `goaccess goaccess` causing it to think 'goaccess' was the log file, and yes that was the case. I deleted the goaccess in start of the command and it fixed it.

I ended up with:
`command: ["--no-global-config", "--config-file=/goaccess/goaccess.conf", "--real-time-html"]`
